frontend/src/scenes/notebooks/Nodes/NotebookNodeImage.tsx

```diff
@@ -0,0 +1,102 @@
+import { api } from '@posthog/apps-common'
+import { NodeViewProps } from '@tiptap/core'
+import { LemonBanner } from 'lib/lemon-ui/LemonBanner'
+import { lazyImageBlobReducer } from 'lib/lemon-ui/LemonFileInput/LemonFileInput'
+import { SpinnerOverlay } from 'lib/lemon-ui/Spinner'
+import { ReactEventHandler, useEffect, useMemo, useState } from 'react'
+import { createPostHogWidgetNode } from 'scenes/notebooks/Nodes/NodeWrapper'
+import { MediaUploadResponse, NotebookNodeType } from '~/types'
+
+const MAX_DEFAULT_HEIGHT = 1000
+
+async function uploadFile(file: File): Promise<MediaUploadResponse> {
+    if (!file.type.startsWith('image/')) {
+        throw new Error('File is not an image')
+    }
+
+    const compressedBlob = await lazyImageBlobReducer(file)
+    const fileToUpload = new File([compressedBlob], file.name, { type: compressedBlob.type })
+
+    const formData = new FormData()
+    formData.append('image', fileToUpload)
+    const media = await api.media.upload(formData)
+
+    return media
+}
+
+const Component = (props: NodeViewProps): JSX.Element => {
+    const { file, src, height } = props.node.attrs
+    const [uploading, setUploading] = useState(false)
+    const [error, setError] = useState<string>()
+
+    useEffect(() => {
```

Comment:
Is there a reason this has to be done in the component? Feels like we could remove the `file` attribute from the component if we uploaded the image and retrieved the `src` before ever creating a node

Reply:
In the future this allows us to manage the "unload" state stopping you from navigating away before the file is finished uploading as well as retries etc.